tests/std/tests/P1614R2_spaceship/test.cpp

```diff
@@ -207,6 +230,94 @@ void diagnostics_test() {
     }
 }
 
+template <template <class...> class TupleLike>
+void tuple_like_test() {
+    {
+        constexpr TupleLike<int, int> t1{1, 1};
+        constexpr TupleLike<int, int> t1_equal{1, 1};
+        constexpr TupleLike<int, int> t2{2, 1};
+
+        static_assert(spaceship_test<std::strong_ordering>(t1, t1_equal, t2));
+    }
+    {
+        TupleLike<int, int> t1{1, 1};
+        TupleLike<int, int> t1_equal{1, 1};
+        TupleLike<int, int> t2{2, 1};
+
+        spaceship_test<std::strong_ordering>(t1, t1_equal, t2);
+    }

```

Comment:
Borderline style comment: You could reduce a lot of duplication here by following our conventional "test both runtime and compiletime" idiom: make `tuple_like_test` `constexpr` and delete 235-241 and 249-255. Then replace all calls to `tuple_like_test<meow>()`; with `tuple_like_test<meow>()`; static_assert((tuple_like_test<meow>(), true));`. (Or have `tuple_like_test` return `true` so you can avoid the comma expression hack in the `static_assert`.)